packages/react-devtools-scheduling-profiler/src/SchedulingProfiler.js

```diff
@@ -7,53 +7,87 @@
  * @flow
  */
 
+import type {Resource} from 'react-devtools-shared/src/devtools/cache';
 import type {ReactProfilerData} from './types';
+import type {ImportWorkerOutputData} from './import-worker/import.worker';
 
 import * as React from 'react';
-import {useState} from 'react';
-
-import ImportButton from './ImportButton';
-import {ModalDialog} from 'react-devtools-shared/src/devtools/views/ModalDialog';
+import {Suspense, useCallback, useState} from 'react';
+import {createResource} from 'react-devtools-shared/src/devtools/cache';
 import ReactLogo from 'react-devtools-shared/src/devtools/views/ReactLogo';
 
+import ImportButton from './ImportButton';
 import CanvasPage from './CanvasPage';
+import ImportWorker from './import-worker/import.worker';
 
 import profilerBrowser from './assets/profilerBrowser.png';
 import styles from './SchedulingProfiler.css';
 
-export function SchedulingProfiler(_: {||}) {
-  const [profilerData, setProfilerData] = useState<ReactProfilerData | null>(
-    null,
-  );
+type DataResource = Resource<void, File, ReactProfilerData | Error>;
```

Comment:
Also not sure if storing data + import errors in a resource is the cleanest approach. I'm doing this to handle these expected/intentional errors without throwing them and using error boundaries. Any idea how I can improve this?